nrf_rpc/doc/architecture.rst

```diff
@@ -3,6 +3,10 @@
 Architecture
```

Comment:
api.rst in this folder also needs the ToC.

Reply:
fixed